tasks/runtime.yml

```diff
@@ -95,11 +95,15 @@
     state: "{{ 'started' if openhpc_slurm_service_started | bool else 'stopped' }}"
   when: openhpc_enable.database | default(false) | bool
 
-# In case the service isn't running and the config hasn't changed to trigger
-# the handler, ensure it's running here.
+- meta: flush_handlers # guarantees order
+
+# Ensure service is running or restarted
 - name: Configure Slurm service
   service:
     name: "{{ openhpc_slurm_service }}"
     enabled: "{{ openhpc_slurm_service_enabled | bool }}"
-    state: "{{ 'started' if openhpc_slurm_service_started | bool else 'stopped' }}"
+    state: |-
+      {% if hostvars[ansible_play_hosts | first]['openhpc_slurm_restart'] | bool -%}restarted
```

Comment:
Sorry, not sure I understand this condition:
hostvars[ansible_play_hosts | first]

Why/how is that node special here?

Reply:
We need a node to set the flag on, but how (else?) can we know the hostname to check whether the flag's been set later?

Reply:
I hadn't managed to find where you set the, looking there...